src/insertItem.ts

```diff
@@ -152,18 +152,63 @@ export class InsertItem {
         };
         let defaultValue = await this.ui.showInputBox({ prompt: "Default value? Leave empty for no default value.", });
         if (defaultValue) {
-            parameter.defaultValue = defaultValue;
+            parameter.defaultValue = this.getDefaultValue(defaultValue, parameterType.value, context);
         }
         let description = await this.ui.showInputBox({ prompt: "Description? Leave empty for no description.", });
         if (description) {
             parameter.metadata = {
                 description: description
             };
         }
-        await this.insertInObject(template, textEditor, templateKeys.parameters, parameter, name, context);
+        await this.insertInObject(template, textEditor, templateKeys.parameters, parameter, name, context, false);
     }
 
-    private async insertInObject(template: DeploymentTemplate, textEditor: vscode.TextEditor, part: string, data: Data | unknown, name: string, context: IActionContext): Promise<void> {
+    private throwError(message: string, context: IActionContext, suppressReportIssue: boolean = true): void {
+        context.errorHandling.suppressReportIssue = true;
+        throw new Error(message);
+    }
+
+    private getDefaultValue(defaultValue: string, parameterType: string, context: IActionContext): string | number | boolean | undefined {
+        switch (parameterType) {
+            case "int":
+                let intValue = Number(defaultValue);
+                if (isNaN(intValue)) {
+                    this.throwError("Invalid integer!", context);
+                }
+                return intValue;
+            case "array":
+                try {
+                    // tslint:disable-next-line: no-unsafe-any
```

Comment:
get rid of as many no-unsafe-any's as you can

Reply:
Fixed.